source/common/regular-expressions.js

```diff
@@ -43,6 +228,117 @@ module.exports = {
       'i' + flag
     )
   },
+
+  /**
+   * Returns a regular expression that matches inline Maths.
+   *
+   * @return  {RegExp}              The compiled Regular Expression
+   */
+  'getInlineMathRE': function () {
+    return RegExp(
+      /^(?:\${1,2}[^\s\\]\${1,2}(?!\d)|\${1,2}[^\s].*?[^\s\\]\${1,2}(?!\d))/.source
+    )
+  },
+
+  /**
+   * Returns a regular expression that matches MarkDown links.
+   *
+   * @return  {RegExp}              The compiled Regular Expression
+   */
+  'getLinkRE': function () {
+    return RegExp(
+      /^.+\.[a-z0-9]+/.source,
+      'i')
+  },
+
+  /**
+   * Returns a regular expression that matches MarkDown lists.
+   *
+   * @return  {RegExp}              The compiled Regular Expression
+   */
+  'getListRE': function () {
+    return RegExp(
+      /^(\s*)([*+-] \[[x ]\]\s|[*+-]\s|(\d+)([.)]))(\s*)/.source
+    )
+  },
+
+  /**
+   * Returns a regular expression that matches empty MarkDown lists.
+   *
+   * @return  {RegExp}              The compiled Regular Expression
+   */
+  'getListEmptyRE': function () {
+    return RegExp(
+      /^(\s*)([*+-] \[[x ]\]|[*+-]|(\d+)[.)])(\s*)$/.source
+    )
+  },
+
+  /**
+   * Returns a regular expression that matches MarkDown Ordered lists.
+   *
+   * @return  {RegExp}              The compiled Regular Expression
+   */
+  'getListOrderedRE': function () {
+    return RegExp(
+      /^(\s*)((\d+)([.)]))(\s*)/.source
+    )
+  },
+
+  /**
+   * Returns a regular expression that matches MarkDown task lists.
+   *
+   * @return  {RegExp}              The compiled Regular Expression
+   */
+  'getListTaskListRE': function () {
+    return RegExp(
+      /^(\s*)(- \[[x ]\])(\s*)/.source
+    )
+  },
+
+  /**
+   * Returns a regular expression that matches unordered Markdown lists.
+   *
+   * @return  {RegExp}              The compiled Regular Expression
+   */
+  'getListTokenRE': function () {
+    return RegExp(
+      /^(\s*)(>[> ]*|[*+-] \[[x ]\]|[*+-]|(\d+)[.)])(\s*)$/.source
+    )
+  },
+
+  /**
+   * Returns a regular expression that matches unordered MarkDown lists.
+   *
+   * @return  {RegExp}              The compiled Regular Expression
+   */
+  'getListUnorderedRE': function () {
+    return RegExp(
+      /[*+-]\s/.source
+    )
+  },
+
+  /**
+   * Returns a regular expression that matches unordered MarkDown lists.
+   *
+   * @return  {RegExp}              The compiled Regular Expression
+   */
+  'getListUnorderedCMRE': function () {
```

Comment:
CM?

Reply:
Codemirror. 
This RE is only used by codemirror stuff. It could probably be consolidated at some point, as there's another RE that does pretty much the same thing, but I don't want to change things until I know this hasn't broken anything.